Cards/CardContainer.gd

```diff
@@ -66,24 +66,41 @@ func set_queued_card(card: CardWorld) -> void:
 	queued_card = card
 
 
-func remove_queued_card() -> void:
+# Remove card from focused and queued and remove it from the hand to prepare it to be played
+func queued_for_active() -> void:
 	_focused_card = null
-	discard_card(queued_card)
-	set_queued_card(null)
+	_remove_queued_card_from_hand()
+
+
+func _remove_queued_card_from_hand() -> void:
+	var _index_to_remove = cards_in_hand.find(queued_card)
+	cards_in_hand.remove_at(_index_to_remove)
 
 
 func is_card_queued() -> bool:
 	return queued_card != null
 
 
-func set_active_card(card: CardBase) -> void:
+func set_active_card(card: CardWorld) -> void:
 	_active_card = card
 
 
-func remove_active_card(card: CardBase) -> void:
+func finish_active_card_action(card: CardBase) -> void:
+	_discard_active_card()
 	_active_card = null
 
 
+func _discard_active_card() -> void:
+	if (_active_card != null):
+		# Add active_card to discard queued to trigger the discard animation
+		_cards_queued_for_discard.append(_active_card)
+		_handle_discard_queue()
```

Comment:
Could maybe just call `_add_to_discard_queue(_active_card)` instead of these two statements

Reply:
OH that's helpful. Thank you for the rec